src/backend/distributed/utils/node_metadata.c

```diff
@@ -1331,9 +1331,15 @@ DeleteNodeRow(char *nodeName, int32 nodePort)
 	HeapTuple heapTuple = NULL;
 	SysScanDesc heapScan = NULL;
 	ScanKeyData scanKey[2];
-
 	Relation pgDistNode = heap_open(DistNodeRelationId(), RowExclusiveLock);
 
+	/*
+	 * simple_heap_delete() expects that the caller has at least an
```

Comment:
What about the other usages of `simple_heap_delete()` on Citus code-base, do you think do we need the same pattern there as well?

As far as I remember, replica identity index is the primary key or the unique index (if explicitly specified) on the table, and all other tables that we do `simple_heap_delete` also operates on tables with pkey or unique index. So, to make the code more roboust, shall we do it for all? If we do, we should make sure to skip incase of `invalidOid`s?

 In that sense, maybe use `GetRelationIdentityOrPK()` function?


 (I guess the reason that only this crashes with pg 12 is that Pkeys are automatically set as the replica identity, but not unique indexes.)

Reply:
There are places where Postgres complains that our catalog tables don't have a replica identity. Should we change the unique indexes to primary keys in SQL?

Reply:
> There are places where Postgres complains that our catalog tables don't have a replica identity. Should we change the unique indexes to primary keys in SQL?

@marcocitus @onderkalaci  Agreed, but since we are tight on time for this release, what do you think about postponing this to 9.0?

I added the lock acquisition for two cases in which we are doing delete from tables with primary key.